src/wayland/shell/xdg/mod.rs

```diff
@@ -1478,6 +1478,15 @@ impl std::cmp::PartialEq for PopupSurface {
     }
 }
 
+impl std::cmp::Eq for PopupSurface {}
+
+impl std::hash::Hash for PopupSurface {
+    fn hash<H: std::hash::Hasher>(&self, state: &mut H) {
+        // self.alive().hash(state);
```

Comment:
```suggestion
```

Reply:
This matches above where it is commented out in the `PartialEq` implementation. Since `Hash` has to be kept in sync with the `PartialEq` implementation, either the commented out part should be entirely removed, or this should remain in case it is un-commented-out.

Reply:
having dead commented out code looks like an artifact, so it must be removed.